Modules/Filtering/ImageIntensity/include/itkHistogramMatchingImageFilter.hxx

```diff
@@ -109,9 +130,17 @@ template< typename TInputImage, typename TOutputImage, typename THistogramMeasur
 const typename HistogramMatchingImageFilter< TInputImage, TOutputImage, THistogramMeasurement >
 ::InputImageType *
 HistogramMatchingImageFilter< TInputImage, TOutputImage, THistogramMeasurement >
-::GetReferenceImage()
+::GetReferenceImage() const
 {
-  return dynamic_cast< TInputImage * >( this->ProcessObject::GetInput(1) );
+  if (m_GenerateReferenceHistogramFromImage)
+    {
+    constexpr unsigned int reference_object_input_idx=1;
```

Comment:
This variable is unnecessary. If you wanted to force 1 to be unsigned, you could have used `1u`, e.g. `GetInput(1u)`.

Reply:
Removed.